main.cpp

```diff
@@ -81,62 +79,24 @@ send me a DM to check your pull request
 If you need to view an example, see: https://bitbucket.org/MatkatMusic/pfmcpptasks/src/master/Projects/Project4/Part7Example.cpp
 */
 
-
-
-
-
-
 #include <iostream>
 #include <cmath>
 #include <functional>
+#include <memory>
+#include <limits>
+#include <typeinfo>
 
-struct DoubleType;
-struct IntType;
-struct FloatType;
 
-struct Point
+template <typename T>
+struct Numeric
 {
-    Point( float a, float b ) : x( a ), y( b ) {}
-    Point( const FloatType& ft );
-    Point( const DoubleType& dt );
-    Point( const IntType& it );
-
-    Point& multiply(float m)
-    {
-        x *= m;
-        y *= m;
-        return *this;
-    }
-    Point& multiply( const FloatType& ft );
-    Point& multiply( const DoubleType& dt );
-    Point& multiply( const IntType& it );
-
-    Point& operator*=( float m )
-    {
-        return multiply( m );
-    }
-
-    void toString()
-    {
-        std::cout << "The x coordinate is: " << x << "\nThe y coordinate is: " << y << std::endl;
-    }
-
-private:
-    float x{0}, y{0};
-};
+    using Primitive = T;
 
-struct FloatType
-{
-    FloatType( float varA ) : a( new float(varA) ) {}
-    ~FloatType()
-    {
-        delete a;
-        a = nullptr;
-    }
+    Numeric( Primitive varA ) : a( new Primitive(varA) ) {}
 
-    operator float() const { return *a; }
+    operator Primitive() const { return *a; }
 
-    FloatType& apply( std::function< FloatType&( float& )> func )
+    Numeric& apply( std::function< Numeric&( Primitive& )> func )
```

Comment:
I missed this in the previous project part:
```
 4) Make both of the member functions's Callable Function Parameter use your owned object as it's single parameter.
         e.g. if you manage your owned object via std::unique_ptr<T>, you'd use this for your std::function argument:
             std::function< OwnedT&(std::unique_ptr<T>&)>
```
- [ ] revise
note: this will mess up your lambdas, and you'll need to consult step 4's hint.